python/ray/serve/_private/proxy.py

```diff
@@ -365,7 +335,108 @@ def _ongoing_requests_end(self):
         self._ongoing_requests -= 1
         self.num_ongoing_requests_gauge.set(self._ongoing_requests)
 
-    async def proxy_request(self, proxy_request: ProxyRequest) -> ProxyResponse:
+    def _get_response_handler_info(
+        self, proxy_request: ProxyRequest
+    ) -> ResponseHandlerInfo:
+        if proxy_request.is_route_request:
+            if self._is_draining():
+                return ResponseHandlerInfo(
+                    response_generator=self.draining_response(proxy_request),
+                    application_name="",
+                    deployment_name="",
+                    route=proxy_request.route_path,
+                    should_record_access_log=False,
+                    should_record_request_metrics=False,
+                    should_increment_ongoing_requests=False,
+                )
+            else:
+                return ResponseHandlerInfo(
+                    response_generator=self.routes_response(proxy_request),
+                    application_name="",
+                    deployment_name="",
+                    route=proxy_request.route_path,
+                    should_record_access_log=False,
+                    # TODO(edoakes): should we be recording metrics for routes?
```

Comment:
well, it's not the same request as everything else because it's a system-level route used for health checking etc.

I think it's fine to record it. just need to make sure we're filtering these out in our dashboards for success rate and such

Reply:
I don't think i see why we'd want to customize its handling from the service perspective, compared to just filtering out these metrics at the time of use.

One point to keep these would be troubleshooting (when you want to validate that service receives the health-checks)

Reply:
@alexeykudinkin Are you asking to log or not to log the metrics? We are already logging them at the moment. The only case we don't is when it's draining. Or are you saying we should log when it's draining, too?